circleci: failed tests: TestMerge

The following test appears to have failed:

[#16125](https://circleci.com/gh/cockroachdb/cockroach/16125):

```
I160411 14:03:11.295004 storage/replica.go:1922  [node=1,store=1,range=1] gossiping system config from store 1, range 1, hash da39a3ee5e6b4b0d3255bfef95601890afd80709
I160411 14:03:11.296016 storage/replica.go:1872  [node=1,store=1,range=1] gossiping sentinel from store 1, range 1
I160411 14:03:11.296291 storage/replica.go:1878  [node=1,store=1,range=1] gossiping first range from store 1, range 1
I160411 14:03:11.297320 storage/replica.go:1409  reproposing 1 commands after empty entry
I160411 14:03:11.302612 storage/engine/rocksdb.go:178  closing in-memory rocksdb instance
--- FAIL: TestMerge (0.04s)
	replica_test.go:3495: Get did not return expected value: aabcd != abcd
=== RUN   TestTruncateLog
W160411 14:03:11.320450 gossip/gossip.go:887  not connected to cluster; use --join to specify a connected node
I160411 14:03:11.321471 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160411 14:03:11.325265 raft/raft.go:463  [group 1] 1 became follower at term 5
I160411 14:03:11.327745 raft/raft.go:238  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I160411 14:03:11.327854 raft/raft.go:543  [group 1] 1 is starting a new election at term 5
I160411 14:03:11.327937 raft/raft.go:476  [group 1] 1 became candidate at term 6
I160411 14:03:11.328013 raft/raft.go:525  [group 1] 1 received vote from 1 at term 6
I160411 14:03:11.328107 raft/raft.go:504  [group 1] 1 became leader at term 6
--
I160411 14:03:53.717393 storage/raft_transport.go:197  establishing Raft transport stream to node 2 at 127.0.0.1:58159
--- PASS: TestInOrderDelivery (0.67s)
=== RUN   Example_rebalancing
W160411 14:03:54.349593 gossip/gossip.go:887  not connected to cluster; use --join to specify a connected node
--- PASS: Example_rebalancing (0.52s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	47.805s
=== RUN   TestBatchBasics
I160411 14:04:06.785348 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160411 14:04:06.787656 storage/engine/rocksdb.go:178  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchGet
I160411 14:04:06.800007 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160411 14:04:06.801383 storage/engine/rocksdb.go:178  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.02s)
=== RUN   TestBatchMerge
I160411 14:04:06.816734 storage/engine/rocksdb.go:137  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

@mrtracy, @bdarnell Is this failure an artifact of the recent change to remove replay protection from `merge` operations at the RocksDB level? Not much in the logs:

```
=== RUN   TestMerge
W160411 14:03:11.286462 gossip/gossip.go:887  not connected to cluster; use --join to specify a connected node
I160411 14:03:11.286690 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160411 14:03:11.289364 raft/raft.go:463  [group 1] 1 became follower at term 5
I160411 14:03:11.289507 raft/raft.go:238  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I160411 14:03:11.289566 raft/raft.go:543  [group 1] 1 is starting a new election at term 5
I160411 14:03:11.289628 raft/raft.go:476  [group 1] 1 became candidate at term 6
I160411 14:03:11.289682 raft/raft.go:525  [group 1] 1 received vote from 1 at term 6
I160411 14:03:11.289755 raft/raft.go:504  [group 1] 1 became leader at term 6
I160411 14:03:11.291995 storage/replica.go:1863  [node=1,store=1,range=1] gossiping cluster id {"bd5acf7c-128f-4a79-8a04-1204f80559b4"} from store 1, range 1
I160411 14:03:11.294725 storage/replica_command.go:1396  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 1s
I160411 14:03:11.295004 storage/replica.go:1922  [node=1,store=1,range=1] gossiping system config from store 1, range 1, hash da39a3ee5e6b4b0d3255bfef95601890afd80709
I160411 14:03:11.296016 storage/replica.go:1872  [node=1,store=1,range=1] gossiping sentinel from store 1, range 1
I160411 14:03:11.296291 storage/replica.go:1878  [node=1,store=1,range=1] gossiping first range from store 1, range 1
I160411 14:03:11.297320 storage/replica.go:1409  reproposing 1 commands after empty entry
I160411 14:03:11.302612 storage/engine/rocksdb.go:178  closing in-memory rocksdb instance
--- FAIL: TestMerge (0.04s)
	replica_test.go:3495: Get did not return expected value: aabcd != abcd
```

--- a/storage/replica_test.go
+++ b/storage/replica_test.go
@@ -3636,10 +3636,9 @@ func TestReplicaStatsComputation(t *testing.T) {
 	verifyRangeStats(tc.engine, tc.rng.RangeID, expMS, t)
 }
 
-// TestMerge verifies that the Merge command is behaving as
-// expected. Merge semantics for different data types are tested more
-// robustly at the engine level; this test is intended only to show
-// that values passed to Merge are being merged.
+// TestMerge verifies that the Merge command is behaving as expected. Time
+// series data is used, as it is the only data type currently fully supported by
+// the merge command.
 func TestMerge(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	tc := testContext{}
@@ -3647,12 +3646,33 @@ func TestMerge(t *testing.T) {
 	defer tc.Stop()
 
 	key := []byte("mergedkey")
-	stringArgs := []string{"a", "b", "c", "d"}
-	stringExpected := "abcd"
+	args := make([]roachpb.InternalTimeSeriesData, 3)
+	expected := roachpb.InternalTimeSeriesData{
+		StartTimestampNanos: 0,
+		SampleDurationNanos: 1000,
+		Samples:             make([]roachpb.InternalTimeSeriesSample, len(args)),
+	}
 
-	for _, str := range stringArgs {
-		mergeArgs := internalMergeArgs(key, roachpb.MakeValueFromString(str))
+	for i := 0; i < len(args); i++ {
+		sample := roachpb.InternalTimeSeriesSample{
+			Offset: int32(i),
+			Count:  1,
+			Sum:    float64(i),
+		}
+		args[i] = roachpb.InternalTimeSeriesData{
+			StartTimestampNanos: expected.StartTimestampNanos,
+			SampleDurationNanos: expected.SampleDurationNanos,
+			Samples:             []roachpb.InternalTimeSeriesSample{sample},
+		}
+		expected.Samples[i] = sample
+	}
 
+	for _, arg := range args {
+		var v roachpb.Value
+		if err := v.SetProto(&arg); err != nil {
+			t.Fatal(err)
+		}
+		mergeArgs := internalMergeArgs(key, v)
 		if _, pErr := tc.SendWrapped(&mergeArgs); pErr != nil {
 			t.Fatalf("unexpected error from Merge: %s", pErr)
 		}
@@ -3668,12 +3688,13 @@ func TestMerge(t *testing.T) {
 	if resp.Value == nil {
 		t.Fatal("GetResponse had nil value")
 	}
-	a, err := resp.Value.GetBytes()
-	if err != nil {
+
+	var actual roachpb.InternalTimeSeriesData
+	if err := resp.Value.GetProto(&actual); err != nil {
 		t.Fatal(err)
 	}
-	if e := []byte(stringExpected); !bytes.Equal(a, e) {
-		t.Errorf("Get did not return expected value: %s != %s", string(a), e)
+	if !proto.Equal(&actual, &expected) {
+		t.Errorf("Get did not return expected value: %v != %v", actual, expected)
 	}
 }
 

